tests/e2e/account/pw_register_test.py

```diff
@@ -0,0 +1,215 @@
+) -> None:
+    assert (
+        page.wait_for_selector('.alert__text').inner_text()
+        == "You are now one of us. Welcome aboard!"
+    )
+
+
+@given("Twoflower visitor is on the home page")
+def when_twoflower_visits_homepage(
+    live_server, page: Page, db_session, user_twoflower_with_password_and_contact
+) -> None:
+    page.goto(live_server.url)
+    db_session.commit()
+
+
+@when("they navigate to the login page")
+def when_navigate_to_login_page(app, live_server, page: Page):
+    page.click('.header__button')
+
+
+@when("they submit the email address with password")
+@when("submit an email address with password")
+def when_submit_email_password(page: Page) -> None:
+    wait_until_recaptcha_loaded(page)
+    page.wait_for_selector('input[name=username]').fill(TWOFLOWER_EMAIL)
+    page.click('#use-password-login')
+    page.wait_for_selector('input[name=password]').fill(TWOFLOWER_PASSWORD)
+    page.click('#login-btn')
+
+
+@then("they are logged in")
+def then_logged_in(live_server, page: Page) -> None:
+    assert (
+        page.wait_for_selector('.alert__text').inner_text() == "You are now logged in"
+    )
+
+
+@when("they submit the phone number with password")
+@when("submit a phone number with password")
+def when_submit_phone_password(app, live_server, page: Page) -> None:
+    wait_until_recaptcha_loaded(page)
+    page.wait_for_selector('input[name=username]').fill(TWOFLOWER_PHONE)
+    page.click('#use-password-login')
+    page.wait_for_selector('input[name=password]').fill(TWOFLOWER_PASSWORD)
+    page.click('#login-btn')
+
+
+@given("Anonymous visitor is on a project page")
+def given_anonymous_project_page(
+    live_server, page: Page, db_session, published_project
+) -> None:
+    page.goto(published_project.absolute_url)
+
+
+@when("they click on follow")
+def when_they_click_follow(page: Page) -> None:
+    page.wait_for_selector("#register-nav").click()
+
+
+@then("a register modal appears")
+def then_register_modal_appear(page: Page) -> None:
+    assert (
+        page.wait_for_selector('xpath=//*[@id="passwordform"]/p[2]').inner_text()
+        == "Tell us where you’d like to get updates. We’ll send an OTP to confirm."
+    )
+
+
+@when(
+    parsers.re("they enter (?P<phone_or_email>a phone number|an email address)"),
+    target_fixture='anon_username',
+)
+def when_they_enter_email(page: Page, phone_or_email: str) -> dict[str, str]:
+    wait_until_recaptcha_loaded(page)
+    if phone_or_email == "a phone number":
+        username = '8123456789'
+    elif phone_or_email == "an email address":
+        username = 'anon@example.com'
+    else:
+        pytest.fail("Unknown username type")
+    page.wait_for_selector('input[name=username]').fill(username)
+    page.click('#form-passwordlogin button')
+    return {'phone_or_email': phone_or_email, 'username': username}
+
+
+@given("Twoflower is on the project page")
+def given_twoflower_visits_project(
+    live_server,
+    page: Page,
+    db_session,
```

Comment:
This doesn't strictly matter, but db_session should be first, then database fixtures, then the server, then the browser.

I remember there was an issue with fixture order in one pytest-bdd test. Reordering fixed it.